### PR TITLE
Run command to check ascds_env in test/env setup

### PR DESCRIPTION
**Pull request by jeanconn**
Run command to check ascds_env in test/env setup

### CODE REVIEW

**Comment by jeanconn** [comment]
This is an idea for #9 
Not sure if it should be a PR on master or py3 at this point (started from py3).

File: agasc/tests/test_agasc.py [hunk 1]
```diff
@@ -37,9 +37,15 @@
     assert os.path.exists(ascrc_file)
     ascds_env = Ska.Shell.getenv('source {} -r release'.format(ascrc_file), shell='tcsh')
     assert 'ASCDS_BIN' in ascds_env
+    # Make a mp_get_agasc cmd line that just returns that command's status
+    cmd = 'mp_get_agasc -r 10 -d 20 -w 0.01 >& /dev/null; echo $?'
+    lines = Ska.Shell.tcsh(cmd, env=ascds_env)
+    # Check command status is 0
+    assert lines[0] == "0"
```
Comment (jeanconn) [inline agasc/tests/test_agasc.py]: Do we have something nicer in the Ska.Shell tools to deal with status?  I thought we must but then couldn't figure it out.

Comment (jeanconn) [inline agasc/tests/test_agasc.py]: I don't immediately see what we gain by making this work with Spawn.  It also looks like the "native" status check in run_shell wants the cmd to return a prompt, and by the time I construct that shell cmd to do that, it seems like explicitly returning the status as done here is basically equivalent.

Comment (taldcroft) [inline agasc/tests/test_agasc.py]: try / except with `run_shell`?
```
In [14]: Ska.Shell.run_shell('FAIL -r 10 -d 20 -w 0.01', env=ascds_env)
---------------------------------------------------------------------------
ShellError                                Traceback (most recent call last)
<ipython-input-14-d7a4bff156e8> in <module>()
----> 1 Ska.Shell.run_shell('FAIL -r 10 -d 20 -w 0.01', env=ascds_env)

/proj/sot/ska/arch/x86_64-linux_CentOS-5/lib/python2.7/site-packages/Ska.Shell-3.3.1-py2.7.egg/Ska/Shell/shell.pyc in run_shell(cmdstr, shell, logfile, importenv, getenv, env)
    174             if exitstatus > 0:
    175                 raise ShellError('Shell command %s failed with exit status %d'
--> 176                                  % (cmdstr, exitstatus))
    177 
    178     # Update os.environ based on changes to environment made by cmdstr

ShellError: Shell command FAIL -r 10 -d 20 -w 0.01 failed with exit status 127

In [15]: Ska.Shell.run_shell('mp_get_agasc -r 10 -d 20 -w 0.01', env=ascds_env)
Out[15]: ([], {})
```

Comment (jeanconn) [inline agasc/tests/test_agasc.py]: I can debug more.  I was getting nothing thrown when running on quango
```
In [6]: cmd = 'mp_get_agasc -r 10 -d 20 -w 0.01'

In [7]: lines = Ska.Shell.tcsh(cmd, env=ascds_env)

In [8]: lines
Out[8]: ['/home/ascds/DS.release/bin/mp_get_agasc: line 81: /home/ascds/DS.release/bin/mp_get_agasc_exe: cannot execute binary file']
```

Comment (jeanconn) [inline agasc/tests/test_agasc.py]: Response in another window suggested I'd be getting non-zero status
```
ascds-quango: mp_get_agasc -r 10 -d 20 -w 0.01
/home/ascds/DS.release/bin/mp_get_agasc: line 81: /home/ascds/DS.release/bin/mp_get_agasc_exe: cannot execute binary file
ascds-quango: echo $?
126
```
but when I tried to set a breakpoint in run_shell to check, I didn't seem to get into the block at https://github.com/sot/Ska.Shell/blob/25fa2476e548bad25c284c76b8ef9005e96575c3/Ska/Shell/shell.py#L164
and then I lost interest.

**Comment by taldcroft** [comment]
Fails (correctly) for me on quango:
```
In [1]: import Ska.Shell

In [2]: ascds_env = Ska.Shell.getenv('source {} -r release'.format('/home/aldcroft/.ascrc'), shell='tcsh')

In [3]: Ska.Shell.run_shell('mp_get_agasc -r 10 -d 20 -w 0.01', env=ascds_env)
---------------------------------------------------------------------------
ShellError                                Traceback (most recent call last)
<ipython-input-3-d76837f71b17> in <module>()
----> 1 Ska.Shell.run_shell('mp_get_agasc -r 10 -d 20 -w 0.01', env=ascds_env)

/proj/sot/ska/arch/i686-linux_CentOS-5/lib/python2.7/site-packages/Ska.Shell-3.3.1-py2.7.egg/Ska/Shell/shell.pyc in run_shell(cmdstr, shell, logfile, importenv, getenv, env)
    174             if exitstatus > 0:
    175                 raise ShellError('Shell command %s failed with exit status %d'
--> 176                                  % (cmdstr, exitstatus))
    177 
    178     # Update os.environ based on changes to environment made by cmdstr

ShellError: Shell command mp_get_agasc -r 10 -d 20 -w 0.01 failed with exit status 126
```

**Comment by jeanconn** [comment]
Oh interesting!  I assumed `Ska.Shell.tcsh` would be equivalent in throwing an exception here because it looks to me like it just calls `run_shell`, but they are behaving differently, so that's a subtlety I do not yet understand.
```
In [8]: Ska.Shell.run_shell(cmd, env=ascds_env)
---------------------------------------------------------------------------
ShellError                                Traceback (most recent call last)
<ipython-input-8-fbdd5d4a0bcd> in <module>()
----> 1 Ska.Shell.run_shell(cmd, env=ascds_env)

/proj/sot/ska/arch/i686-linux_CentOS-5/lib/python2.7/site-packages/Ska.Shell-3.3.1-py2.7.egg/Ska/Shell/shell.pyc in run_shell(cmdstr, shell, logfile, importenv, getenv, env)
    174             if exitstatus > 0:
    175                 raise ShellError('Shell command %s failed with exit status %d'
--> 176                                  % (cmdstr, exitstatus))
    177 
    178     # Update os.environ based on changes to environment made by cmdstr

ShellError: Shell command mp_get_agasc -r 10 -d 20 -w 0.01 failed with exit status 126

In [9]: lines = Ska.Shell.tcsh(cmd, env=ascds_env)
In [10]: 
```

**Comment by jeanconn** [comment]
And apparently that's due to the shell type

```
In [12]: Ska.Shell.run_shell(cmd, env=ascds_env, shell='tcsh')
Out[12]: 
(['/home/ascds/DS.release/bin/mp_get_agasc: line 81: /home/ascds/DS.release/bin/mp_get_agasc_exe: cannot execute binary file'],
 {})

In [13]: Ska.Shell.run_shell(cmd, env=ascds_env, shell='bash')
---------------------------------------------------------------------------
ShellError                                Traceback (most recent call last)
<ipython-input-13-1eaed3b35cd8> in <module>()
----> 1 Ska.Shell.run_shell(cmd, env=ascds_env, shell='bash')

/proj/sot/ska/arch/i686-linux_CentOS-5/lib/python2.7/site-packages/Ska.Shell-3.3.1-py2.7.egg/Ska/Shell/shell.pyc in run_shell(cmdstr, shell, logfile, importenv, getenv, env)
    174             if exitstatus > 0:
    175                 raise ShellError('Shell command %s failed with exit status %d'
--> 176                                  % (cmdstr, exitstatus))
    177 
    178     # Update os.environ based on changes to environment made by cmdstr

ShellError: Shell command mp_get_agasc -r 10 -d 20 -w 0.01 failed with exit status 126
```

**Comment by taldcroft** [comment]
Argh.  Somehow `tcsh` isn't causing an error as it should.  Anyway just go with your original plan, good enough!

**Comment by jeanconn** [comment]
Nah, I thought your run_shell was a lot cleaner, so I amended the commit to do that anyway.

We might end up with some more special-case Ska.Shell stuff in other checks like this as we improve tests so good to know about the possible difference between the shells (which might be a Ska.Shell bug or a tcsh oddity... but I agree it doesn't really matter just now).

**Pull request merged**